sympy/assumptions/ask.py

```diff
@@ -1495,13 +1653,25 @@ def get_known_facts():
         Equivalent(Q.irrational | Q.rational, Q.real & Q.finite),
         Implies(Q.irrational, ~Q.rational),
         Implies(Q.zero, Q.even),
+        Equivalent(Q.extended_positive, Q.positive | Q.infinite),
+        Equivalent(Q.extended_negative, Q.negative | Q.infinite),
```

Comment:
I think these should be implies rather than equivalent because e.g. `-oo` satisfies `Q.infinite` but not `Q.extended_positive`.

Reply:
Ah, this may be the thing that breaks it! I was a bit confused about how to do this...